project/.github/workflows/ci.yml

```diff
@@ -48,11 +48,16 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
+      - name: Install poetry
+        run: pipx install poetry
+      # https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages
       - name: Set up Python
         uses: actions/setup-python@v5
+        id: setup-python
         with:
           python-version: ${{ matrix.python-version }}
-      - uses: snok/install-poetry@v1.3.4
```

Comment:
I've just realised that this action uses the deprecated install script, and that the solution you suggest is the recommended one 👍